internal/policies/fetcher.go

```diff
@@ -0,0 +1,127 @@
+)
+
+const (
+	kubewardenPoliciesGroup   = "policies.kubewarden.io"
+	kubewardenPoliciesVersion = "v1"
+)
+
+// Fetcher fetches Kubewarden policies from the Kubernetes cluster, and filters policies that are auditable.
+type Fetcher struct {
+	client client.Client
+	filter func(policies []policiesv1.Policy) []policiesv1.Policy
+}
+
+// NewFetcher returns a Fetcher. It will try to use in-cluster config, which will work just if audit-scanner is deployed
+// inside a Pod. If in-cluster fails, it will try to fetch the kube config from the home dir. It will return an error
+// if both attempts fail.
+func NewFetcher() (*Fetcher, error) {
+	client, err := newClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Fetcher{client: client, filter: filterAuditablePolicies}, nil
+}
+
+// TODO implement this for all ns
+func (f *Fetcher) GetPoliciesForAllNamespaces() ([]policiesv1.Policy, error) {
+	return nil, errors.New("scanning all namespaces is not implemented yet. Please pass the --namespace flag to scan a namespace")
+}
+
+// GetPoliciesForANamespace gets all auditable policies for a given namespace
+func (f *Fetcher) GetPoliciesForANamespace(namespace string) ([]policiesv1.Policy, error) {
+	namespacePolicies, err := f.findNamespacesForAllClusterAdmissionPolicies()
+	if err != nil {
+		return nil, fmt.Errorf("can't fetch ClusterAdmissionPolicies: %v", err)
+	}
+	admissionPolicies, err := f.getAdmissionPolicies(namespace)
+	if err != nil {
+		return nil, fmt.Errorf("can't fetch AdmissionPolicies: %v", err)
+	}
+	for _, policy := range admissionPolicies {
+		namespacePolicies[namespace] = append(namespacePolicies[namespace], &policy)
+	}
+
+	return f.filter(namespacePolicies[namespace]), nil
+}
+
+// initializes map with an entry for all namespaces with an empty policies array as value
+func (f *Fetcher) initNamespacePoliciesMap() (map[string][]policiesv1.Policy, error) {
+	namespacePolicies := make(map[string][]policiesv1.Policy)
+	namespaceList := &v1.NamespaceList{}
+	err := f.client.List(context.Background(), namespaceList, &client.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("can't list namespaces: %v", err)
+	}
+	for _, namespace := range namespaceList.Items {
+		namespacePolicies[namespace.Name] = []policiesv1.Policy{}
+	}
+
+	return namespacePolicies, nil
+}
+
+// returns a map with an entry per each namespace. Key is the namespace name, and value is an array of ClusterAdmissionPolicies
+// that will evaluate resources within this namespace.
+func (f *Fetcher) findNamespacesForAllClusterAdmissionPolicies() (map[string][]policiesv1.Policy, error) {
+	namespacePolicies, err := f.initNamespacePoliciesMap()
+	if err != nil {
+		return nil, err
+	}
+	policies := &policiesv1.ClusterAdmissionPolicyList{}
+	err = f.client.List(context.Background(), policies, &client.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("can't list AdmissionPolicies: %v", err)
+	}
+
+	for _, policy := range policies.Items {
+		policy := policy
```

Comment:
```suggestion
```

Reply:
I think this isn't needed. We don't intend to update `policy` meanwhile we are going through the range.

Reply:
yes, we need this. We [append](https://github.com/kubewarden/audit-scanner/blob/a7a141fcfd8d91b287ec4d849f4586e241030e81/internal/policies/fetcher.go#L94) `&policy` which will be updated in each loop iteration

Reply:
doh! :sweat_smile: . Resolving!